crates/burn-tensor/src/tensor/api/base.rs

```diff
@@ -298,6 +298,95 @@ where
+    /// in the tensor; otherwise, the dimension will not be squeezed. If `dims` is empty, all single-dimensional entries
+    /// in the tensor will be removed. If entries in `dims` are negative, then dimensions will be counted
+    /// from the back.
+    ///
+    /// # Arguments
+    ///
+    /// - `dims`: The dimension(s) to be squeezed.
+    ///
+    /// # Type Parameters
+    ///
+    ///  - 'D2': The resulting number of dimensions in the squeezed tensor.
+    ///
+    /// # Returns
+    ///
+    /// A new `Tensor<B, D2, K>` instance with the specified dimensions removed.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    ///
+    /// use burn_tensor::backend::Backend;
+    /// use burn_tensor::{Tensor, Shape};
+    ///
+    /// fn example<B: Backend>() {
+    ///     let device = Default::default();
+    ///     let tensor = Tensor::<B, 4>::ones(Shape::new([2, 1, 4, 1]), &device);
+    ///
+    ///     // Given a 4D tensor with dimensions (2, 1, 4, 1), squeeze the 1 and 3 dimensions
+    ///     let squeezed_tensor: Tensor::<B, 2> = tensor.squeeze_dims(&[1, 3]);
+    ///
+    ///     // Resulting tensor will have dimensions (2, 4)
+    ///     println!("{:?}", squeezed_tensor.shape());
+    /// }
+    /// ```
+    pub fn squeeze_dims<const D2: usize>(self, dims: &[isize]) -> Tensor<B, D2, K> {
+        let current_dims = self.shape().dims;
+        let mut dim_indices: Vec<usize>;
+
+        // Check if dims is empty, if yes then assign dim_indices all single-dimensional entries
+        if dims.is_empty() {
+            dim_indices = current_dims
+                .iter()
+                .enumerate()
+                .filter_map(|(index, &dim)| if dim == 1 { Some(index) } else { None })
+                .collect();
+        } else {
+            // If negative dims, count from the back
+            dim_indices = dims
+                .iter()
+                .map(|&d| {
+                    if d < 0 {
+                        (current_dims.len() as isize + d) as usize
+                    } else {
+                        d as usize
+                    }
+                })
+                .collect();
+        }
+
+        // Sort indices and remove duplicates
+        dim_indices.sort_unstable();
+        dim_indices.dedup();
+
+        // Make sure squeeze_dims doesn't result in a tensor with < 1 dimensions
+        check!(TensorCheck::squeeze_dims_input::<D2>(
+            &dim_indices,
+            &current_dims
+        ));
+
+        // Calculate new dimensions
+        let mut new_dims = Vec::new();
+        for (index, &dim_size) in current_dims.iter().enumerate() {
+            // Exclude the dimension if it's explicitly marked for squeezing and is of size 1
+            if dim_indices.contains(&index) && dim_size == 1 {
+                check!(TensorCheck::squeeze::<D2>(index, &current_dims));
+                continue;
+            }
```

Comment:
@nathanielsimard this check doesn't do much of anything right now since the `==1` basically means there's no chance of it erroring out. The `squeeze()` function will panic if you ask it to squeeze a dimension that isn't 1. For this, I'm not sure if you want to keep the same behavior as `squeeze()` or not. It looks like the ONNX spec says it will raise an error if you try squeezing a non-1 dimension. But PyTorch would just leave that dimension unchanged.

Reply:
In my opinion it should behave like our `squeeze`, so panic on non-1 dimensions